src/components/comparison/ComparisonExpandableRow.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <ComparisonRow
+    :studies="studies"
+    :title="title"
+    :subtitle="subtitle"
+    :getValue="getValue"
+    expandable
+    :expanded="expanded"
+    @toggle-expand="toggleExpand()"
+  >
+    <template #default="{ value }">
+      <slot :value="value"></slot>
+    </template>
+  </ComparisonRow>
+  <ComparisonRow
+    v-for="subKey in subKeys"
+    :studies="studies"
+    class="sub-row"
+    :class="{ expanded }"
+    :title="subKey"
+    subtitle="-"
+    :getValue="study => getSubValues(study)[subKey]"
+  >
+    <template #default="{ value }">
+      <slot :value="value"></slot>
+    </template>
+  </ComparisonRow>
+</template>
+
+<script setup>
+import { computed, ref } from 'vue';
+import ComparisonRow from './ComparisonRow.vue';
+
+const props = defineProps({
+    studies: Array,
+    title: String,
+    subtitle: String,
+    getValue: Function,
+    getSubValues: Function,
```

Comment:
Je me demande si ça ne serai pas plus simple à utiliser en passant directement des objects bien formattés contenant les valeurs à affiché, plutôt que des fonctions que le composant doit appeler. Et il faudrait peut-être ajouter quelque part une description de ce à quoi l'objet (ou le retour de la fonction) doit ressembler (propriétés obligatoires, leur type, etc...).
Mais c'est juste un questionnement et un 'nice to have' dont on peut se passer.

Reply:
@leonarf C'est une bonne idée ! J'avoue etre tenté par des props du style `values` et `subValues`, qui feraient que les composants parents gereraient tout :thinking: Je ne sais pas laquelle des 2 solutions est la meilleure, mais c'est assez facile d'en changer a partir de cette implem en tout cas.

En général, je suis un peu frileux des gros objets qu'on passe et qui ne sont pas très concrets pour les composants qui transmettent cette info (ex si on passait `valuesWithSubValues = [{ value: 3, subValues: { Producers: 1, Wholesalers: 2} }, { value: 9, ...}, ...]`, ca ferait quelque chose d'un peu gros.

A voir, je garde ça en tête et si je vois une occasion de changer ça je le fais.

Reply:
@leonarf 

> Et il faudrait peut-être ajouter quelque part une description de ce à quoi l'objet (ou le retour de la fonction) doit ressembler (propriétés obligatoires, leur type, etc...).

Avec des props typées, ça marche du tonerre, ça donnerait ça

```ts
const props = defineProps<{
    studies: StudyData[]; // Tableau de "StudyData", type défini pour l'ensemble du code pour la consistence
    title: string;
    subtitle: string;
    getValue: (study: StudyData) => number; // Fonction qui prends comme argument `StudyData` et return un nombre
    getSubValues: (study: StudyData) => Record<StageName, number | null>; // Renvoie un object, dont les clés sont des noms de stage ("Producers" | "Collectors" |...), et le résultat est un nombre ou `null`
}>();